uwblueprint/project-read-backend
# Safe delete fields

### Notion ticket link



[Soft delete fields](https://www.notion.so/uwblueprintexecs/Soft-delete-fields-6f59dd0337fe480ab609b6e6be65a2f9)



### Implementation description

- Added [safe delete package ](https://github.com/makinacorpus/django-safedelete)
- By default fields are now soft deleted. There is a date time field called `deleted` on the field model which indicates when a field was marked as deleted. They are not actually deleted from the db.
- we can undelete fields via the Django admin or we could add an endpoint to undelete them



### Steps to test

1. Build and run the container `docker-compose up -d --build`
2. Login to the admin
3. Go to `localhost:8000/fields/{id}` and pick whatever id you like (as long as it exists)
4. Delete the field
5. Go back to the admin and check that the field you deleted was highlighted as deleted
6. Try undeleting it from admin (select and undelete from admin actions)
![image](https://user-images.githubusercontent.com/21024100/132110858-fe36c2c9-726a-446a-9bda-462e97f70c80.png)



### What should reviewers focus on?

- testing

### Checklist
